app/controllers/users/two_factor_verification_controller.rb

```diff
@@ -0,0 +1,57 @@
+module Users
+  class TwoFactorVerificationController < ApplicationController
+    before_action :check_direct_otp_expiry, only: [:show, :create]
+
+
+    def show
+      unless current_user.totp_enabled?
+        current_user.create_direct_otp
+        r = SMSService.new.send_two_factor_confirmation_code!(current_user)
+
+        # TODO: make this async and handle service failure
+        unless r.code == 202
+          raise 'There was an error sending the SMS'
+        end
+      end
+    end
+
+    def create
```

Comment:
This comes down to how we expect the SMS service to work. It's currently modelled off the `two_factor_authentication` gem which essentially sends it on `#show`.

This has an advantage in that it doesn't require the user have to click through to have an SMS sent to them - however the disadvantage is it doesn't provide the user with a warning that 2FA will be required, or give them an opportunity to back out of it.

We could choose to use the show action to display an informative screen outlining that verification is required via 2FA, and get the user to click through (post to create) to send the SMS. The subsequent call to submit the code could be an update action.